hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4214,10 +4214,40 @@ func (r *HostedClusterReconciler) reconcileAWSSubnets(ctx context.Context, creat
 	return nil
 }
 
+func releaseImageToVersion(hcluster *hyperv1.HostedCluster, image string) (semver.Version, error) {
```

Comment:
My preference would be to just look up the version from the release image. For example, see:
https://github.com/openshift/hypershift/blob/04a743410f373e2421813e9935f419db8832c7f0/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L1398-L1405

Reply:
So the thing is that the provider is a controller-level variable while the pull secret we use is hostedcluster-specific. We do want to use the pull secret that is provided with the HostedCluster if at some point you are trying to use a release that only your pull secret has access to.